format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -69,7 +69,48 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
         if (delegate.nextKeyValue()) {
           // skip to next if the current record is header
           if (skipHeader && delegate.getCurrentKey().get() == 0L) {
-            return delegate.nextKeyValue();
+            delegate.nextKeyValue();
```

Comment:
need to return false if this returns false. Otherwise it will error on files that only contain a header

Reply:
condition added.